lib/worker.js

```diff
@@ -299,44 +298,36 @@ class AssetComputeWorker {
                 this.renditions[rendition.index] = newRendition;
                 rendition = newRendition;
 
-            } catch (err) {
-                // if postprocessing fails, rendition will be failed too
+            } else {
                 this.timers.postProcessing += postProcessingTimer.duration();
-                console.error(`post-processing ${rendition.name} failed after ${postProcessingTimer.durationSec()}:`, err);
-
-                await this.renditionFailure(rendition, new GenericError(`Post-processing of image rendition failed`, "image_post_process"));
+            }
 
-                // return undefined to mark as failed
-                return undefined;
+            return rendition;
 
-            }
-        } else {
+        } catch (err) {
+            // if postprocessing fails, rendition will be failed too
             this.timers.postProcessing += postProcessingTimer.duration();
-        }
+            console.error(`post-processing ${rendition.name} failed after ${postProcessingTimer.durationSec()}:`, err);
 
-        return rendition;
-    }
+            await this.renditionFailure(rendition, new GenericError(`Post-processing of image rendition failed`, "image_post_process"));
+
+            // return undefined to mark as failed
+            return undefined;
 
-    shouldPostProcess(rendition) {
-        if (!rendition.postProcess) {
-            return false;
         }
+    }
 
-        // verify if the requested format is one we can output from post-processing
-        if (!["png", "jpg", "jpeg", "gif", "tif", "tiff"].includes(rendition.instructions.fmt)) {
-            console.log(`Skipping post-processing because fmt ${rendition.instructions.fmt} is not supported`);
+    async shouldPostProcess(rendition) {
+        if (!rendition.postProcess) {
```

Comment:
Didn't we want to remove that flag and leave the decision entirely to the SDK?

Reply:
I think it would be nice to keep a flag that ensures it goes straight to post-processing. This would save time for `worker-flite` that already does the check inside the worker to see if it should do post-processing. No, need to examine the file twice.

Reply:
Correct, opt-in. And in the flite special case we always need a way to tell the post processing to not run... although if the logic is right, the decision in sdk should also figure out it should not run. But since we already made the check in flite, it's good to not repeat it and e.g. inspect the intermediate rendition unnecessarily.